routing/routing.go

```diff
@@ -175,17 +225,59 @@ func (r *Routing) startReceivingUpdates(o Options) {
 	}()
 }
 
-// Matches a request in the current routing tree.
+// Route matches a request in the current routing tree.
 //
 // If the request matches a route, returns the route and a map of
 // parameters constructed from the wildcard parameters in the path
 // condition if any. If there is no match, it returns nil.
 func (r *Routing) Route(req *http.Request) (*Route, map[string]string) {
-	m := r.matcher.Load().(*matcher)
-	return m.match(req)
+	rt := r.routeTable.Load().(*routeTable)
+	return rt.m.match(req)
 }
 
-// Closes routing, stops receiving routes.
+// Close closes routing, stops receiving routes.
 func (r *Routing) Close() {
 	close(r.quit)
 }
+
+func slice(r []*eskip.Route, offset int, limit int) []*eskip.Route {
+	if offset > len(r) {
+		offset = len(r)
+	}
+	end := offset + limit
+	if end > len(r) {
+		end = len(r)
+	}
+	result := r[offset:end]
+	if result == nil {
```

Comment:
I think this condition will never happen.

[2] go-pry> a = []string{"foo"}
=> []string{"foo"}
[1] go-pry> a[0:0]
=> []string{}
[4] go-pry> a[0:0] == nil
=> false